zebrad/src/components/tokio.rs

```diff
@@ -57,12 +57,12 @@ impl RuntimeRun for Runtime {
             }
         });
 
+        // Enforce shutdown by avoiding long blocking tasks
+        self.shutdown_timeout(std::time::Duration::from_secs(5));
```

Comment:
I am not totally sure but i extended to 20 seconds here and i am not getting more C++ failures by now ... so it maybe needs more time to finish ? One issue could be that the database is still writing a batch and needs time to completes that work.

Reply:
Calling `cancel_all_background_work(true)` should wait for all the RocksDB tasks to exit. So it could also be tokio that needs more time to shut down.

I increased the tokio timeout to 20 seconds in my PR https://github.com/oxarbitrage/zebra/pull/164